eth/integrity/rcache_no_duplicates.go

```diff
@@ -0,0 +1,213 @@
+package integrity
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"sync/atomic"
+	"time"
+
+	"github.com/erigontech/erigon-db/rawdb"
+	"github.com/erigontech/erigon-lib/common"
+	"github.com/erigontech/erigon-lib/kv"
+	"github.com/erigontech/erigon-lib/log/v3"
+	"github.com/erigontech/erigon/turbo/services"
+	"golang.org/x/sync/errgroup"
+)
+
+func CheckRCacheNoDups(ctx context.Context, db kv.TemporalRoDB, blockReader services.FullBlockReader, failFast bool) (err error) {
+	defer func() {
+		log.Info("[integrity] RCacheNoDups: done", "err", err)
+	}()
+
+	logEvery := time.NewTicker(10 * time.Second)
+	defer logEvery.Stop()
+
+	txNumsReader := blockReader.TxnumReader(ctx)
+
+	tx, err := db.BeginTemporalRo(ctx)
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
+
+	rcacheDomainProgress := tx.Debug().DomainProgress(kv.RCacheDomain)
+	fromBlock := uint64(1)
+	toBlock, _, _ := txNumsReader.FindBlockNum(tx, rcacheDomainProgress)
+
+	{
+		log.Info("[integrity] RCacheNoDups starting", "fromBlock", fromBlock, "toBlock", toBlock)
+		accProgress := tx.Debug().DomainProgress(kv.AccountsDomain)
+		if accProgress != rcacheDomainProgress {
+			err := fmt.Errorf("[integrity] RCacheDomain=%d is behind AccountDomain=%d", rcacheDomainProgress, accProgress)
+			log.Warn(err.Error())
+			return err
+		}
+	}
+
+	tx.Rollback()
+
+	defer db.Debug().EnableReadAhead().DisableReadAhead()
```

Comment:
my idea was: to enable only MADV_NORMAL or MADV_RANDOM
because: MADV_SEQUENTIAL is very special thing - it aggressively evict everything what behind current read. And we have many accessors - with non-sequential access, etc...
MADV_NORMAL - works similar to MADV_SEQUENTIAL for sequential reads - it has growing-window of pre-fetch if OS see that successfully predicted reads.